src/glaciers/data/SurfaceVelocityData.jl

```diff
@@ -7,36 +7,38 @@ by providing `nothing` as the default value.
 `SurfaceVelocityData{F <: AbstractFloat} <: AbstractData`
 
 # Fields
- - `x::Union{Vector{F}, Nothing}`: Easting of observation.
- - `y::Union{Vector{F}, Nothing}`: Northing of observation.
- - `lat::Union{Vector{F}, Nothing}`: Latitude of observation.
- - `lon::Union{Vector{F}, Nothing}`: Longitude of observation.
- - `vx::Union{Array{F, 3}, Nothing}`: x component of surface velocity.
- - `vy::Union{Array{F, 3}, Nothing}`: y component of surface velocity.
- - `vabs::Union{Array{F, 3}, Nothing}`: Absolute ice surface velocity.
- - `vx_error::Union{Array{F, 1}, Nothing}`: Error in `vx`
- - `vy_error::Union{Array{F, 1}, Nothing}`: Error in `vy`
- - `vabs_error::Union{Array{F, 1}, Nothing}`: Error in `vabs`.
- - `date::Union{Vector{DateTime}, Nothing}`: Date of observation (mean of `date1` and `date2`)
- - `date1::Union{Vector{DateTime}, Nothing}`: First date of adquisition.
- - `date2::Union{Vector{DateTime}, Nothing}`: Second date of adquisition.
- - `date_error::Union{Vector{Day}, Vector{Millisecond}, Nothing}`: Error in `date`.
+- `x::Union{Vector{F}, Nothing}`: Easting of observation.
+- `y::Union{Vector{F}, Nothing}`: Northing of observation.
+- `lat::Union{Vector{F}, Nothing}`: Latitude of observation.
+- `lon::Union{Vector{F}, Nothing}`: Longitude of observation.
+- `vx::Union{Vector{Matrix{F}}, Nothing}`: x component of surface velocity.
+- `vy::Union{Vector{Matrix{F}}, Nothing}`: y component of surface velocity.
+- `vabs::Union{Vector{Matrix{F}}, Nothing}`: Absolute ice surface velocity.
+- `vx_error::Union{Vector{F}, Nothing}`: Error in `vx`
+- `vy_error::Union{Vector{F}, Nothing}`: Error in `vy`
+- `vabs_error::Union{Vector{F}, Nothing}`: Error in `vabs`.
+- `date::Union{Vector{DateTime}, Nothing}`: Date of observation (mean of `date1` and `date2`)
+- `date1::Union{Vector{DateTime}, Nothing}`: First date of adquisition.
+- `date2::Union{Vector{DateTime}, Nothing}`: Second date of adquisition.
+- `date_error::Union{Vector{Day}, Vector{Millisecond}, Nothing}`: Error in `date`.
+- `glacierGridded::Bool`: Whether the data have been gridded to the glacier grid or not.
 """
 mutable struct SurfaceVelocityData{F <: AbstractFloat} <: AbstractData
     x::Union{Vector{F}, Nothing}
     y::Union{Vector{F}, Nothing}
     lat::Union{Vector{F}, Nothing}
     lon::Union{Vector{F}, Nothing}
-    vx::Union{Array{F, 3}, Nothing}
-    vy::Union{Array{F, 3}, Nothing}
-    vabs::Union{Array{F, 3}, Nothing}
-    vx_error::Union{Array{F, 1}, Nothing}
-    vy_error::Union{Array{F, 1}, Nothing}
-    vabs_error::Union{Array{F, 1}, Nothing}
+    vx::Union{Vector{Matrix{F}}, Nothing}
+    vy::Union{Vector{Matrix{F}}, Nothing}
+    vabs::Union{Vector{Matrix{F}}, Nothing}
```

Comment:
This looks good to me! Just curious: any reason why a vector of matrices is better than just a 3D tensor?

Reply:
I think it's nicer to handle. Since two dimensions are spatial and one is temporal, I feel it's much nice to iterate through time and get back spatial grids. Otherwise the user needs to know which of the 3 dimensions is time.